pandas/core/frame.py

```diff
@@ -8190,11 +8190,16 @@ def update(
         if not isinstance(other, DataFrame):
             other = DataFrame(other)
 
-        other = other.reindex_like(self)
+        # reindex rows, non-matching columns get skipped
+        other = other.reindex(self.index)
 
         for col in self.columns:
+            if col not in other.columns:
```

Comment:
Could probably do a union of `self.columns.union(other.columns)` first and iterate over that instead of constantly doing `in`